records/generator.go

```diff
@@ -27,9 +29,17 @@ type rrs map[string][]string
 // RecordGenerator contains DNS records and methods to access and manipulate
 // them. TODO(kozyraki): Refactor when discovery id is available.
 type RecordGenerator struct {
-	As       rrs
-	SRVs     rrs
-	SlaveIPs map[string]string
+	As         rrs
+	SRVs       rrs
+	SlaveIPs   map[string]string
+	httpClient http.Client
+}
+
+// NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
+func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
```

Comment:
why pass in the timeout here? probably better off passing in a http.Client since there's likely to be other things that will want configuration in the future (e.g. SSL settings)